File: IOTManagementGroup7.Models/AirConditioner.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Data;
using System.Diagnostics;
using System.Dynamic;
using System.Text;

namespace IOTManagementGroup7.Models
{
	public class AirConditioner
	{

		[Key]
		public int Id { get; set; }
		[Required]
		public string ApplicationUserId { get; set; }
		[ForeignKey("ApplicationUserId")]
		public ApplicationUser ApplicationUser { get; set; }

		public string Name { get; set; }
		public bool PowerStatus { get; set; }
		public float Temperature { get; set; }
		public string SourceCode { get; set; }
	}
}

File: IOTManagementGroup7/wwwroot/js/airConditioner.js
var dataTable;

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    dataTable = $("#tblData").DataTable({
        "ajax": {
            "url": "/Admin/AirConditioner/GetAll"
        },
        "columns": [
            { "data": "applicationUser.userName", "width": "15%" },
            { "data": "name", "width": "15%" },
            { "data": "powerStatus", "width": "15%" },
            { "data": "temperature", "width": "15%" },
            { "data": "sourceCode", "width": "20%" },

            {
                "data": "id",
                "render": function (data) {
                    return `
                            <div class="text-center">
                                <a href="/Admin/AirConditioner/Upsert/${data}" class="btn btn-success text-white" style="cursor:pointer">
                                    <i class="fas fa-edit"></i>
                                </a>
                                <a onclick=Delete("/Admin/AirConditioner/Delete/${data}") class="btn btn-danger text-white" style="cursor:pointer">
                                    <i class="fas fa-trash-alt"></i> 
                                </a>
                            </div>
                            `;
                }, "width": "15%"
            }
        ]
    })
}

function Delete(url) {
    swal({
        title: "Are you sure you want to Delete?",
        text: "You will not be able to restore the data!",
        icon: "warning",
        buttons: true,
        dangerMode: true
    }).then((willDelete) => {
        if (willDelete) {
            $.ajax({
                type: "DELETE",
                url: url,
                success: function (data) {
                    if (data.success) {
                        toastr.success(data.message);
                        dataTable.ajax.reload();
                    }
                    else {
                        toastr.error(data.message);
                    }
                }
            });
        }
    });
}